src/main/java/ch/njol/skript/util/SoundUtils.java

```diff
@@ -1,49 +1,65 @@
-/**
- *   This file is part of Skript.
- *
- *  Skript is free software: you can redistribute it and/or modify
- *  it under the terms of the GNU General Public License as published by
- *  the Free Software Foundation, either version 3 of the License, or
- *  (at your option) any later version.
- *
- *  Skript is distributed in the hope that it will be useful,
- *  but WITHOUT ANY WARRANTY; without even the implied warranty of
- *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- *  GNU General Public License for more details.
- *
- *  You should have received a copy of the GNU General Public License
- *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
- *
- * Copyright Peter Güttinger, SkriptLang team and contributors
- */
 package ch.njol.skript.util;
 
+import ch.njol.skript.Skript;
+import org.bukkit.NamespacedKey;
 import org.bukkit.Sound;
 import org.jetbrains.annotations.Nullable;
 
-/**
- * @author Peter Güttinger
- */
-public abstract class SoundUtils {
-	private SoundUtils() {}
-	
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Locale;
+
+public class SoundUtils {
+
+	private static final boolean IS_INTERFACE;
+	private static final Method VALUE_OF_METHOD;
+	private static final Method GET_KEY_METHOD;
+
 	static {
-		assert false;
+		try {
+			Class<?> SOUND_CLASS = Class.forName("org.bukkit.Sound");
+			IS_INTERFACE = SOUND_CLASS.isInterface();
+			VALUE_OF_METHOD = SOUND_CLASS.getDeclaredMethod("valueOf", String.class);
```

Comment:
Using Enum.valueOf would allow you to avoid reflection for this method

Reply:
I'll do some testing with that and update short... stay tuned!

Reply:
Ok, I think I got it working, fingers crossed tests pass